site/app/controllers/student/SubmissionController.php

```diff
@@ -1390,7 +1391,128 @@ private function updateSubmissionVersion() {
 
         return array('error' => false, 'version' => $new_version, 'message' => $msg);
     }
-    
+
+    private function ajaxUploadImagesFiles() {
+        if($this->core->getUser()->getGroup() === 4) {
+			     return $this->uploadResult("You have no permission to access this page", false);
+        }
+
```

Comment:
Also, one needs instructor permissions to view the upload images button, so I think it would make sense to need the same level of permissions to upload images, not just grader level.